src/HasParsedMessage.php

```diff
@@ -27,8 +28,9 @@ trait HasParsedMessage
      */
     public function date(): ?CarbonInterface
     {
-        if ($date = $this->header(HeaderConsts::DATE)?->getDateTime()) {
-            return Carbon::instance($date);
+        $dateHeader = $this->header(HeaderConsts::DATE);
```

Comment:
That's what I mean. That's my prefered approach. Runtime checking for the right instance